substrate/frame/referenda/src/lib.rs

```diff
@@ -499,6 +506,12 @@ pub mod pallet {
 				alarm: Self::set_alarm(nudge_call, now.saturating_add(T::UndecidingTimeout::get())),
 			};
 			ReferendumInfoFor::<T, I>::insert(index, ReferendumInfo::Ongoing(status));
+			let (status, track) = Self::as_ongoing(index)
+				.expect("has been added to ReferendumInfoFor as ongoing; qed");
```

Comment:
🤯 You're right! The `status` that's here is not actually a status, but an instance of `TallyOf<T>`.

What I can do to avoid this redundant read (since the tally has just been created anyways, is to input a copy of the created tally).